src/base/bittorrent/infohash.h

```diff
@@ -38,15 +43,44 @@ using SHA256Hash = Digest32<256>;
 
 namespace BitTorrent
 {
-    class InfoHash : public SHA1Hash
+    class InfoHash;
+
+    class TorrentID : public SHA1Hash
     {
     public:
         using SHA1Hash::SHA1Hash;
 
-        static InfoHash fromString(const QString &hashString);
+        static TorrentID fromString(const QString &hashString);
+        static TorrentID fromInfoHash(const InfoHash &infoHash);
     };
 
-    uint qHash(const InfoHash &key, const uint seed);
+    class InfoHash
+    {
+    public:
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+        using UnderlyingType = lt::info_hash_t;
+#else
+        using UnderlyingType = lt::sha1_hash;
+#endif
+
+        InfoHash() = default;
+        InfoHash(const InfoHash &other) = default;
+        InfoHash(const UnderlyingType &nativeHash);
+
+        bool isValid() const;
+        TorrentID toTorrentID() const;
+
+        operator UnderlyingType() const;
```

Comment:
Can't it be private? or do you intend to expose it to public?

Reply:
>you intend to expose it to public?

Yes.